tests/infra/logging_app.py

```diff
@@ -2,60 +2,117 @@
         LOG.success(f"Applying {number_txs} logging txs")
         primary, backup = network.find_primary_and_any_backup()
         remote_node = backup if on_backup else primary
 
+        self.issue_on_node(network, remote_node, number_txs, consensus)
+
+    def issue_on_node(self, network, remote_node, number_txs, consensus):
+        LOG.success(f"Applying {number_txs} logging txs to node {remote_node.node_id}")
         with remote_node.node_client() as mc:
             check_commit = infra.checker.Checker(mc)
             check_commit_n = infra.checker.Checker(mc, self.notifications_queue)
 
             with remote_node.user_client() as uc:
-                for _ in range(number_txs):
-                    priv_msg = f"Private message at index {self.next_priv_index}"
-                    pub_msg = f"Public message at index {self.next_pub_index}"
-                    rep_priv = uc.rpc(
-                        "LOG_record", {"id": self.next_priv_index, "msg": priv_msg,},
-                    )
-                    rep_pub = uc.rpc(
-                        "LOG_record_pub", {"id": self.next_pub_index, "msg": pub_msg,},
-                    )
-                    check_commit_n(rep_priv, result=True)
-                    check_commit(rep_pub, result=True)
-
-                    self.priv[self.next_priv_index] = priv_msg
-                    self.pub[self.next_pub_index] = pub_msg
-                    self.next_priv_index += 1
-                    self.next_pub_index += 1
-
-        network.wait_for_node_commit_sync(consensus)
+                for _ in range(self.tx_index_start, self.tx_index_start + number_txs):
+                    end_time = time.time() + self.timeout
+                    while time.time() < end_time:
+                        try:
+                            priv_msg = (
+                                f"Private message at index {self.next_priv_index}"
+                            )
+                            pub_msg = f"Public message at index {self.next_pub_index}"
+                            rep_priv = uc.rpc(
+                                "LOG_record",
+                                {"id": self.next_priv_index, "msg": priv_msg,},
+                            )
+                            rep_pub = uc.rpc(
+                                "LOG_record_pub",
+                                {"id": self.next_pub_index, "msg": pub_msg,},
+                            )
+                            check_commit_n(rep_priv, result=True)
+                            check_commit(rep_pub, result=True)
+
+                            self.priv[self.next_priv_index] = priv_msg
+                            self.pub[self.next_pub_index] = pub_msg
+                            self.next_priv_index += 1
+                            self.next_pub_index += 1
+                            break
+                        except (TimeoutError, requests.exceptions.ReadTimeout,) as e:
```

Comment:
`requests.exceptions.ReadTimeout` implies that this wouldn't work with the `CurlClient`?